examples/platforms/simulation/virtual_time/platform-sim.c

```diff
@@ -108,13 +113,27 @@ static void receiveEvent(otInstance *aInstance)
     case OT_SIM_EVENT_ALARM_FIRED:
         break;
 
+    case OT_SIM_EVENT_UART_WRITE:
+        otPlatUartReceived(event.mData, event.mDataLength);
+        break;
+
+#if OPENTHREAD_SIMULATION_EXT_RF_MODELS
+    case OT_SIM_EVENT_RADIO_RX_INTERFERED:
     case OT_SIM_EVENT_RADIO_RECEIVED:
-        platformRadioReceive(aInstance, event.mData, event.mDataLength);
+        platformRadioReceive(aInstance, event.mData, event.mDataLength, event.mParam1);
         break;
 
-    case OT_SIM_EVENT_UART_WRITE:
-        otPlatUartReceived(event.mData, event.mDataLength);
+    case OT_SIM_EVENT_RADIO_TX_DONE:
+        // the external RF simulator determines success/error of Tx, and must report an otError code.
+        assert(event.mDataLength >= 1);
+        otError err = (otError)event.mData[0];
+        platformRadioTransmitDone(aInstance, err);
+        break;
+#else
+    case OT_SIM_EVENT_RADIO_RECEIVED:
```

Comment:
Ok, that sounds like a better approach. In this case if a user tries to run an "old" simulator with a "new" OT binary there could be a failure message (telling to upgrade the simulator) and the OT node can stop.

Reply:
If the same OT binary does not have to support both old and new OTNS / python-simulator, that would imply we can remove the compilation flag OPENTHREAD_SIMULATION_EXT_RF_MODELS .  Instead, the code would then default to the code (as if OPENTHREAD_SIMULATION_EXT_RF_MODELS =1 1) and the code for old simulator support (OPENTHREAD_SIMULATION_EXT_RF_MODELS == 0 would be removed. 
@simonlingoogle  Did you have this approach in mind? It would improve code readability.

Reply:
> If the same OT binary does not have to support both old and new OTNS / python-simulator, that would imply we can remove the compilation flag OPENTHREAD_SIMULATION_EXT_RF_MODELS . Instead, the code would then default to the code (as if OPENTHREAD_SIMULATION_EXT_RF_MODELS =1 1) and the code for old simulator support (OPENTHREAD_SIMULATION_EXT_RF_MODELS == 0 would be removed. @simonlingoogle Did you have this approach in mind? It would improve code readability.

Agree that we should default to `OPENTHREAD_SIMULATION_EXT_RF_MODELS=1` to improve readability and maintainability. 

We generally don't need to worry about interops between old/new simulator and new/old OT simulations.